src/utils/isStatelessComponent.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+import isReactModuleName from './isReactModuleName';
+import isReactComponentClass from './isReactComponentClass';
+import isReactCreateClassCall from './isReactCreateClassCall';
+import match from './match';
+import recast from 'recast';
+import resolveToModule from './resolveToModule';
+import resolveToValue from './resolveToValue';
+
+var {types: {namedTypes: types}} = recast;
+
+const validPossibleStatelessComponentTypes = [
+  'Property',
+  'FunctionDeclaration',
+  'FunctionExpression',
+  // TODO: maybe include these variants for safety:
+  // https://github.com/benjamn/ast-types/blob/master/def/es6.js#L31
+  // https://github.com/estree/estree/issues/2
+  // 'ArrowExpression', 'ArrowFunction'
+  'ArrowFunctionExpression'
+];
+
+function containsJSXElementOrReactCreateElementCall(node) {
+  var visited = false;
+  recast.visit(node, {
+    visitJSXElement(path) {
+      visited = true;
```

Comment:
Yeah, I’ve thought of this as well. After I get through the current cases with detecting propTypes (last step I believe) then I’ll look into maybe doing something like a return value lookup of each function rather than the current heuristic of “contains JSX or React.createElement”. I suspect this could get a bit tricky or annoying or untenable if return value is something imported from another module, though. I intend on not supporting that :)